pkg/rancher-desktop/backend/containerClient/__tests__/registry.spec.ts

```diff
@@ -0,0 +1,43 @@
+import { FetchError } from 'node-fetch';
+
+import dockerRegistry, { getAsList } from '@pkg/backend/containerClient/registry';
+import { Headers } from '@pkg/utils/fetch';
+
+describe('Headers', () => {
+  test('getAsList', () => {
+    const headers = new Headers();
+
+    headers.append('a', '1');
+    headers.append('A', '2');
+    headers.append('b', '3');
+
+    expect(headers.get('a')).toEqual('1, 2');
+    expect(headers[getAsList]('a')).toEqual(['1', '2']);
+    expect(headers[getAsList]('B')).toEqual(['3']);
+    expect(headers[getAsList]('c')).toBeNull();
+  });
+});
+
+describe('DockerRegistry', () => {
+  describe('getTags', () => {
+    it('should get tags from unauthenticated registry', async() => {
+      const reference = 'registry.opensuse.org/opensuse/leap';
+
+      await expect(dockerRegistry.getTags(reference))
```

Comment:
Should we be mocking `fetch` here or is it ok in this case to make network calls in a unit test?

Reply:
While I prefer mocking `fetch`, that's a lot of code just to mock a registry. This one is less of an issue, but the Docker Hub one (which requires getting a token) is the real issue.